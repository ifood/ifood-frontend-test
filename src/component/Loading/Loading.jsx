import React from 'react';
import { Dimmer, Loader } from 'semantic-ui-react';
import { FormattedMessage } from 'react-intl';

export default function Loading() {
  return (
    <Dimmer active inverted>
      <Loader inverted>
        <FormattedMessage id="loading" />
      </Loader>
    </Dimmer>
  );
}
