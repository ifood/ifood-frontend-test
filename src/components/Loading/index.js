import React from "react";

import { Spinner, BounceOne, BounceTwo, BounceThree } from "./styles";

const Loading = () => {
  return (
    <Spinner>
      <BounceOne />
      <BounceTwo />
      <BounceThree />
    </Spinner>
  );
};

export default Loading;
