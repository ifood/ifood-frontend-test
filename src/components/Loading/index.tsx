import React from 'react';

import Logo from '../Logo';

import { Container, Progress } from './styles';

const Loading = () => (
  <Container>
    <Logo width="230px" color="red" />
    <Progress />
  </Container>
);

export default Loading;
