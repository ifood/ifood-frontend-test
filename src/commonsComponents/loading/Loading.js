import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const useStyles = makeStyles(({
  root: {
    display: 'flex',
    justifyContent: 'center',
    width: '100%',
    alignItems: 'center',
    marginTop: '10vh'
  },
}));

export const Loading = () =>  {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <CircularProgress />
    </div>
  );
}
