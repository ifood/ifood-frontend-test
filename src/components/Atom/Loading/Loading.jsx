import React from 'react'
import PropTypes from 'prop-types'
import Box from '@material-ui/core/Box'
import CircularProgress from '@material-ui/core/CircularProgress'
import './Loading.scss'

const Loading = ({ className }) => {
  return (
    <Box className={className}>
      <CircularProgress />
    </Box>
  )
}

Loading.propTypes = {
  className: PropTypes.string,
}

export default React.memo(Loading)
