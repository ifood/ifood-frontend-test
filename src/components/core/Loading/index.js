import React from "react";
import PropTypes from "prop-types";
import Lottie from "react-lottie";

import * as S from "./styles";
import loading from "assets/loading/loading.json";

const Loading = ({ isLoading, children }) => {
  if (isLoading)
    return (
      <S.Container>
        <S.Svg>
          <Lottie
            options={{
              animationData: loading,
            }}
          />
        </S.Svg>
      </S.Container>
    );

  return children;
};

Loading.propTypes = {
  isLoading: PropTypes.bool,
  children: PropTypes.any,
};

Loading.defaultProps = {
  isLoading: false,
  children: null,
};

export default Loading;
