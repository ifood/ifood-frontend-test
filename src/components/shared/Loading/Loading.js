import React from 'react';
import { Col, Spin } from 'antd';
import { LoadingOutlined } from '@ant-design/icons';
import './Loading.less';

const Loading = () => (
  <Col
    span={24}
    className="loading"
  >
    <Spin indicator={(
      <LoadingOutlined
        className="loading__spin"
        spin
      />
  )}
    />
  </Col>
);

export default Loading;
