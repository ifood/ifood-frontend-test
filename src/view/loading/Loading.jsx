import React from 'react';
import { CircularProgress, Typography } from '@material-ui/core';
import { CenteredFlex } from '../../component/containers/Containers';

export default function Loading() {
  return (
    <CenteredFlex>
      <CircularProgress />
      <Typography color="textSecondary">Loading...</Typography>
    </CenteredFlex>
  );
}
