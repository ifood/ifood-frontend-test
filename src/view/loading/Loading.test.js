/* eslint-disable no-undef, react/jsx-filename-extension, react/prop-types */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Loading from './Loading';

describe('<Loading />', () => {
  it('snapshot', () => {
    const wrapper = mount(<Loading />);
    jestExpect(toJson(wrapper)).toMatchSnapshot();
  });
});
