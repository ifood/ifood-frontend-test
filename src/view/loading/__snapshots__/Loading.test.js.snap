// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

<Loading>
  <styled.div>
    <div
      className="c0"
    >
      <WithStyles(ForwardRef(CircularProgress))>
        <ForwardRef(CircularProgress)
          classes={
            Object {
              "circle": "MuiCircularProgress-circle",
              "circleDisableShrink": "MuiCircularProgress-circleDisableShrink",
              "circleIndeterminate": "MuiCircularProgress-circleIndeterminate",
              "circleStatic": "MuiCircularProgress-circleStatic",
              "colorPrimary": "MuiCircularProgress-colorPrimary",
              "colorSecondary": "MuiCircularProgress-colorSecondary",
              "indeterminate": "MuiCircularProgress-indeterminate",
              "root": "MuiCircularProgress-root",
              "static": "MuiCircularProgress-static",
              "svg": "MuiCircularProgress-svg",
            }
          }
        >
          <div
            className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style={
              Object {
                "height": 40,
                "width": 40,
              }
            }
          >
            <svg
              className="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx={44}
                cy={44}
                fill="none"
                r={20.2}
                strokeWidth={3.6}
                style={Object {}}
              />
            </svg>
          </div>
        </ForwardRef(CircularProgress)>
      </WithStyles(ForwardRef(CircularProgress))>
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
      >
        <ForwardRef(Typography)
          classes={
            Object {
              "alignCenter": "MuiTypography-alignCenter",
              "alignJustify": "MuiTypography-alignJustify",
              "alignLeft": "MuiTypography-alignLeft",
              "alignRight": "MuiTypography-alignRight",
              "body1": "MuiTypography-body1",
              "body2": "MuiTypography-body2",
              "button": "MuiTypography-button",
              "caption": "MuiTypography-caption",
              "colorError": "MuiTypography-colorError",
              "colorInherit": "MuiTypography-colorInherit",
              "colorPrimary": "MuiTypography-colorPrimary",
              "colorSecondary": "MuiTypography-colorSecondary",
              "colorTextPrimary": "MuiTypography-colorTextPrimary",
              "colorTextSecondary": "MuiTypography-colorTextSecondary",
              "displayBlock": "MuiTypography-displayBlock",
              "displayInline": "MuiTypography-displayInline",
              "gutterBottom": "MuiTypography-gutterBottom",
              "h1": "MuiTypography-h1",
              "h2": "MuiTypography-h2",
              "h3": "MuiTypography-h3",
              "h4": "MuiTypography-h4",
              "h5": "MuiTypography-h5",
              "h6": "MuiTypography-h6",
              "noWrap": "MuiTypography-noWrap",
              "overline": "MuiTypography-overline",
              "paragraph": "MuiTypography-paragraph",
              "root": "MuiTypography-root",
              "srOnly": "MuiTypography-srOnly",
              "subtitle1": "MuiTypography-subtitle1",
              "subtitle2": "MuiTypography-subtitle2",
            }
          }
          color="textSecondary"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            Loading...
          </p>
        </ForwardRef(Typography)>
      </WithStyles(ForwardRef(Typography))>
    </div>
  </styled.div>
</Loading>
`;
